### ISSUE TITLE
[Basic question] What are the learnable model parameters in Colbert?

### ISSUE THREAD

**Issue by maksimt** [stanford-futuredata/ColBERT#74]
Hi, I'm new to neural nets and pytorch, and after reading your paper+code I wasn't 100% sure what the learnable parameters in ColBERT are. I saw that you were responsive to even basic questions on Github (which is great IMO 👍👍), so I figured I'd ask.

Question 1: based on my understanding, in ColBERT we learn:

1. The linear embedding layer that maps from BERT's output dimension to the dimension used for MaxSim. This is learned from scratch?
2. Next, we fine-tune the pre-trained embeddings for all terms in BERT's vocabulary. This includes "[meta]" used in query augmentation, and however BERT represent "[Q]" and "[D"]. That is to say, we don't learn any new parameters here, rather we fine tune existing BERT parameters to the extractive Q&A datasets?
  
That's it right?

Question 2: in [your paper](https://arxiv.org/pdf/2004.12832.pdf) in Section 3.2, Equations (1) for E_q and (2) for E_d before normalizing, you apply "CNN" to the BERT embeddings. Does this mean convolutional neural nets or something else? Is CNN represented anywhere in the code in [forward](https://github.com/stanford-futuredata/ColBERT/blob/master/colbert/modeling/colbert.py#L33) e.g. for the Query?

**Comment by okhat** [comment]
Thanks for the questions, @maksimt! 

The learnable parameters are all of the BERT ones (including the embedding layer for the vocabulary as well as all of the BERT encoder layers) and also the linear layer, which is learned from scratch indeed.

The CNN in the equation you mention refers to the linear layer. It was initially implemented as a unit-kernelsize CNN when we were exploring various designs.

**Comment by okhat** [comment]
Marking as resolved. Feel free to re-open if you have more questions!
